code/src/reducers/tasks.js

```diff
@@ -0,0 +1,29 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+const tasks = createSlice({
+    name: 'tasks',
+    initialState: {
+        items: [],
+    },
+    reducers: {
+        toggleItem: (store, action) => {      
+            store.items.forEach(item => {
+                if (item.id === action.payload) {
+                    item.complete = !item.complete
+                }
+            })
+        },
+        //Mutable reducer function to use splice and push but works since immer included in dependencies
+        deleteItem: (store, action) => {
+            store.items.splice(action.payload, 1)
+        },
+        addItem: (store, action) => {
+            store.items.push(action.payload)
+        },
+        removeAll: (store) => {
+            store.items = []
+        }
+    }
+})
+
```

Comment:
Very well-written reducers! You passed along the perfect amount of information and named them perfectly. Beautiful to see.

The only reducer I was "missing" was one to toggle all tasks as complete. That could have looked something like this:

```
    toggleAllTasksComplete: (state, action) => {
      const tasks = store.items;
      for (let i = 0; i < tasks.length; i++) {
        items[i].complete = true;
      }
    },
```